src/core/include/openvino/core/extension.hpp

```diff
@@ -24,6 +25,14 @@ class Extension;
  */
 class OPENVINO_API Extension {
 public:
+    _OPENVINO_HIDDEN_METHOD static const DiscreteTypeInfo& get_type_info_static() {
+        static const ::ov::DiscreteTypeInfo type_info_static{"Extension"};
+        return type_info_static;
+    }
+    virtual const DiscreteTypeInfo& get_type_info() const {
+        return get_type_info_static();
+    }
```

Comment:
@t-jankowski I see that in some places we need to use custom RTTI declaration to ensure it's marked as `virtual`

Can we expend `OPENVINO_RTTI` for such cases ? I suppose default `OPENVINO_RTTI` does not provide virtual? should we have something like `OPENVINO_RTTI_BASE_CLASS` or similar ? to denote that we later want to override definitions

Reply:
Yes, `OPENVINO_RTTI_BASE_CLASS` would enforce virtual for `get_type_info`. I'll make it.

Reply:
Addressed in https://github.com/openvinotoolkit/openvino/pull/28911